### PR TITLE
Update dropdowns to extend outside the coding workspace

### PR DESCRIPTION
**Pull request by jmkulwik**
Relates to this PR: https://github.com/droplet-editor/droplet/pull/209
Ticket: https://codedotorg.atlassian.net/browse/STAR-461

Droplet dropdowns now render in front of other parts of the screen and respect their z-index.

In the past, the code workspace had all overflows set to "hidden" this effectively caused the droplet dropdowns to ignore their z-index and hide any section that interacted with another component on the screen. It would appear to render under these components. This "hidden" overflow was used to make the droplet palette "disappear" when the palette was closed because it would interact with elements outside the code workspace. This change removes the "hidden" overflow. Additionally, it adjusts the palette to disappear under the code workspace when closed rather than disappear under the app screen when closed.

**Dropdown before:**
image

**Dropdown after:**
image

**Palette before:**
oldToolbox

**Palette after:**
newToolbox2

**FOR RECORDS (NO LONGER ACCURATE - fixed after comments) Palette after:**
newToolbox

### CODE REVIEW

**Comment by cpirich** [comment]
Removing the `overflow: hidden` may have other side effects in Blockly levels. It would be good to verify that we're not seeing any problems there (which may fairly subtle). I'd check it out carefully in the inspector to see if any `codeWorkspace` changes size at all in Blockly mode when you toggle the `overflow` values. Maybe eyes tests could be used as well?

**Review by cpirich — CHANGES_REQUESTED** [review]
The `after` image above shows the white rectangle of the droplet editor clearly painting outside of its container (overwriting the gray border on the right side). I think this needs to be fixed before we can be comfortable with this approach. We had to use `overflow: hidden` because this component couldn't be trusted to stay in its rectangle. It looks like there may still be cases where that is the case.
The safer fix may be to keep an `overflow: hidden` `div` in the picture for almost all of `droplet`, but not for the dropdowns.

**Comment by jmkulwik** [comment]
> The safer fix may be to keep an `overflow: hidden` `div` in the picture for almost all of `droplet`, but not for the dropdowns.

^ @cpirich Can you expand on this? I'm not completely sure what you mean. 

On that note, one easy-fix possibility is to make overflow-x hidden. I can also look into why it's expanding outside its container in the first place and fix that.

**Comment by cpirich** [comment]
> > The safer fix may be to keep an `overflow: hidden` `div` in the picture for almost all of `droplet`, but not for the dropdowns.
> 
> ^ @cpirich Can you expand on this? I'm not completely sure what you mean.
> 
> On that note, one easy-fix possibility is to make overflow-x hidden. I can also look into why it's expanding outside its container in the first place and fix that.

Right now, most everything within droplet is contained by an `overflow: hidden` `div` (what they call `@wrapperElement`). Rather than remove `overflow: hidden` from that element, you could try to just create the dropdown elements outside of the `@wrapperElement`. This is just a thought - it may not be sufficient to fix the problem that I was just observing. The reason I look for a general, safe fix is that there may be other places where they'll draw outside of the rectangle - so it could take a while to find them all.

File: apps/lib/droplet/droplet-full.js
Comment (joshlory) [inline apps/lib/droplet/droplet-full.js:5]: This changes the date, but no actual code changes to `droplet-full.js`?

Comment (joshlory) [reply apps/lib/droplet/droplet-full.js:5]: Sorry, this comment ends up showing up on the full diff 😄. I'm looking at commit 20988918bfb3d9a43590808cc67be396194198ec specifically.

File: apps/src/StudioApp.js
Comment (joshlory) [inline apps/src/StudioApp.js:2035]: Where does this style get cleared? (so dropdowns don't get clipped after switching to text mode and back)

Comment (jmkulwik) [reply apps/src/StudioApp.js:2035]: Can you clarify?

File: apps/lib/droplet/droplet-full.js
Comment (jmkulwik) [reply apps/lib/droplet/droplet-full.js:5]: Yeah, there was a small coffescript spacing change and a couple test changes. This basically ends up being a no-op commit that records the version of droplet that we are using.

Comment (joshlory) [reply apps/lib/droplet/droplet-full.js:5]: Got it! Thx for clarifying.

File: apps/src/StudioApp.js
Comment (joshlory) [reply apps/src/StudioApp.js:2035]: Followed up offline. My understanding is that there are two different wrappers. We no longer clip on the inner wrapper, but still clip on the outer wrapper. That's probably ok, because users would need a very small screen size to trigger this remaining bug:

image

File: apps/style/applab/style.scss
Comment (joshlory) [inline apps/style/applab/style.scss:491]: 👍 for removing this override, and porting it to the Droplet codebase!

**Review by joshlory — APPROVED** [review]
LGTM! I'll defer to @cpirich if he wants to take one final pass on the issue where he Requested Changes.

File: apps/src/StudioApp.js
Comment (jmkulwik) [reply apps/src/StudioApp.js:2035]: I also don't think I'd consider it a bug. It makes sense for a dropdown like that to display over the instruction space, but when it starts displaying over the nav bar, it starts getting in the way of different types of user actions. i.e. when I'm coding, it makes sense that some items might overflow into the instructions. When I'm trying to go to a different lesson, it doesn't make sense for items from the coding workspace to overflow on the level progression. i.e. the app should be contained by the project space, not displayed on top of the project space

File: apps/style/applab/style.scss
Comment (cpirich) [reply apps/style/applab/style.scss:491]: The reason we did this before is because we wanted 100% of our changes to go back into droplet `master`. If that is no longer a goal, I agree with this change.

Comment (cpirich) [reply apps/style/applab/style.scss:491]: I see that we now believe that these changes are compatible with other consumers of droplet, so it does seems possible to merge our changes into `master` at some point.

**Review by cpirich — APPROVED** [review]
Glad that you were able to take the time to verify the changes with a consumer of droplet and fix the one case where we could see droplet growing outside of the codeWorkspace. We should keep an eye out for any other new issues. Thanks again!

**Pull request merged**
